File: Easy/1.py
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i, num in enumerate(nums):
            pair = target - num
            if pair in dict:
                return [i, dict[pair]]
            dict[num] = i
if __name__ == '__main__': 
    sol = Solution()
    tests  = [([2,7,11,15], 9), ([3,2,4], 6), ([3,3], 6)]
    for nums, target in tests:
        ans = sol.twoSum(nums, target)
        print(ans)

            

File: C++/Easy/1.cc
#include <unordered_map>
#include <vector>

using namespace std;


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); ++i){
            int remain{target - nums[i]};
            // found remain in seen
            if (seen.find(remain) != seen.end()){
                return {seen[remain], i};
            }
            // not found in seen
            seen[nums[i]] = i;
        }
        return {};
    }
};